src/Components/Forms/TestsAndTreatments/Instructions/InstructionsForTreatment.js

```diff
@@ -144,19 +236,113 @@ const InstructionsForTreatment = ({ encounter }) => {
     }
     return clean;
   };
+  const updateEncounterExtension = async (
+    encounter,
+    selectedStatus,
+    practitioner,
+  ) => {
+    try {
+      if (!selectedStatus) return;
+      const cloneEncounter = { ...encounter };
+      switch (selectedStatus) {
+        case 'waiting_for_nurse':
+          cloneEncounter.status = 'in-progress';
+          break;
+        case 'waiting_for_doctor':
+          cloneEncounter.status =
+            cloneEncounter.status === 'arrived' ? 'triaged' : 'in_progress';
+
+          break;
+        case 'waiting_for_xray':
+          cloneEncounter.status = 'in-progress';
+          break;
+      }
+      cloneEncounter.extensionSecondaryStatus = selectedStatus;
+      cloneEncounter.practitioner = practitioner;
+
+      await FHIR('Encounter', 'doWork', {
```

Comment:
According of my suggest above you can remove the lines 263-270 and return cloneEncounter instead

Reply:
see what I wrote above if you can clarify by phone or something I just don't understand ....